### PR TITLE
Implement hourly time resolution

### PR DESCRIPTION
**Pull request by saltaverde**
Description from first commit pasted here for convenience:

```
Change implementation to use hourly time resolution

The initial problem was that some departments have special shifts on
leap year days and these shifts are sometimes of different length than
24 hours. Namely, Delaware, OH has three 8-hour shifts on leap days.

The existing shift patterns were based on the assumption of a 24 hour
period per pattern character. These changes allow for a variable shift
duration via the `shiftDuration` parameter.
```

Closes https://github.com/StatEngine/shiftly/issues/18

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -44,33 +44,32 @@ class ShiftInformation {
...
+      momentDate.add(this.shiftStartDate.hour(), 'hours');
     }
 
-    const patternLength = this.pattern.length;
+    let hoursFromStart = this.hoursFromPatternStart(momentDate);
 
-    const mod = Math.abs(daysFromStart) % patternLength;
-    let shift = null;
+    // Adjust for time spans covering DST transition :(
+    if (this.patternStart.isDST() !== momentDate.isDST()) {
```
Comment (richard3d) [inline src/index.js:61]: I'm not sure we can trust this for all departments. For instance, Any department in Hawaii or Arizona will not recognize DST period. Thoughts?

**Review by richard3d — CHANGES_REQUESTED** [review]
See comment regarding DST. Can we also get an update on the Readme regarding usage and how to account for any things to be aware of in the parameters such as the new `shiftDuration`?

File: src/index.js (hunk 1 above)
Comment (saltaverde) [reply src/index.js:61]: I believe we'd have to specify time zones specifically for those "special" places, like `US/Arizona` as opposed to `US/Mountain`. But should confirm that the behavior is as expected with `moment`

Comment (saltaverde) [reply src/index.js:61]: It appears to me that `moment` understands that certain timezones do not observe the transition to/from DST, `US/Arizona` being one of them:
```
> var azDstDate = moment.tz('2020-11-01 00:00:00', 'US/Arizona');
undefined
> var azNotDstDate = moment.tz('2020-12-01 00:00:00', 'US/Arizona');
undefined
> azDstDate.isDST()
false
> azNotDstDate.isDST()
false
> var mtDstDate = moment.tz('2020-11-01 00:00:00', 'US/Mountain');
undefined
> var mtNotDstDate = moment.tz('2020-12-01 00:00:00', 'US/Mountain');
undefined
> mtDstDate.isDST();
true
> mtNotDstDate.isDST();
false
```
So, I think we're good.

**Pull request merged**